### PR TITLE
Add domain picker to FSE

### PR DESCRIPTION
**Pull request by alshakero**
Depends on https://github.com/Automattic/wp-calypso/pull/43045.

#### Changes

The PR is the first step towards Domain Picker in FSE. Currently the domain picker:

1. Has a working search feature.
2. Perfectly styled.
3. Categories are intact.
4. All Confirm/close buttons work

##### Reading
1. I'm using a global (`_currentSiteId`) and Sites store to get the site title to initiate the domain search. Feels hacky.
2. I'm constructing the `currentDomain` by processing `site.URL`. Feels hacky too and doesn't contain all the domain info (is free, relevance, category, etc..)

##### Writing
When users set a domain (or a plan) we need to use `wpcom.cart` to store that information. There is a good example [here](https://github.com/Automattic/wp-calypso/blob/master/client/landing/gutenboarding/hooks/use-on-site-creation.ts#L101-L105). Something like this:
```js
const cart: Cart = await wpcom.getCart( newSite.site_slug );
await wpcom.setCart( newSite.blogid, {
	...cart,
	products: [ ...cart.products, planProduct, domainProduct ],
} );
```
What's unclear to me is whether this will add two domains to the cart or the backened will remove one of them automatically when you add a new domain product. This will need some investigation.

Also, I'm not sure how to access `wpcom` library in wp-admin, since I think it's part of `client` folder. 

#### Testing

1. In root, run `yarn start` to run Calypso
2. In `apps/full-site-editing`, run `yarn dev --sync` to run the app and sync it with your sandbox.
3. Create and sandbox a simple site.
4. Define a `A8C_FSE_DOMAIN_PICKER_ENABLE` flag and set it to true in PHP.
5. When you land in the editor page, a working domain picker button should be visible.

### CODE REVIEW

**Comment by razvanpapadopol** [comment]
This PR is superseding https://github.com/Automattic/wp-calypso/pull/42834, right?

**Comment by alshakero** [comment]
> This PR is superseding #42834, right?

Sadly I totally forgot about that one. Luckily, there is a little overlap at this point so I can steal Yan's work into mine :D

File: apps/full-site-editing/package.json [hunk 1]
```diff
@@ -82,10 +82,12 @@
...
 	},
 	"dependencies": {
-		"@automattic/format-currency": "^1.0.0-alpha.0",
-		"@automattic/data-stores": "^1.0.0-alpha.1",
 		"@automattic/calypso-build": "*",
+		"@automattic/data-stores": "^1.0.0-alpha.1",
+		"@automattic/domain-picker": "1.0.0-alpha.0",
+		"@automattic/format-currency": "^1.0.0-alpha.0",
 		"@wordpress/api-fetch": "*",
+		"@wordpress/base-styles": "1.9.0",
```
Comment (simison) [inline apps/full-site-editing/package.json:90]: Likely needs `^`

Comment (simison) [reply apps/full-site-editing/package.json:90]: PRed https://github.com/Automattic/wp-calypso/pull/43058 since there seems to be NPM config problem.

**Pull request merged**